src/measure_theory/constructions/pi.lean

```diff
@@ -496,6 +499,46 @@ begin
   exacts [ennreal.prod_lt_top (λ i _, hμ i), λ i, (ho i).measurable_set]
 end
 
+variable (μ)
+
+/-- Separating the coordinates into those that satisfy a predicate `p` and those that don't maps
+a product measure to a product of product measures. This is useful to apply Fubini to some subset
+of the variables. The converse is `map_pi_equiv_pi_subtype_prod` -/
+lemma map_pi_equiv_pi_subtype_prod_symm (p : ι → Prop) [decidable_pred p] :
+  map (equiv.pi_equiv_pi_subtype_prod p α).symm
+    (measure.prod (measure.pi (λ i, μ i)) (measure.pi (λ i, μ i))) = measure.pi μ :=
```

Comment:
I assume this works?
```suggestion
    ((measure.pi (λ i, μ i)).prod (measure.pi (λ i, μ i))) = measure.pi μ :=
```

Reply:
No, it doesn't (unification is not easy here because I haven't prescribed the domains of the different `i`s)

Reply:
Fair. Do you happen to know what the difference is for unification between dot notation and regular notation? I've run into this kind of problem before but don't really understand it.

Reply:
I don't really understand the details, but I can imagine what is going on. I guess the issue here is the following: in your suggestion, it has to understand that `.prod` means `measure.prod`. So it tries to find the type of the first `(measure.pi (λ i, μ i))`. But here it has no information on what `i` should be, so it guesses that it is probably just running over `ι`. And then the global formula doesn't make sense. With the explicit `measure.prod`, on the other hand, it doesn't have to find right away the type of `(measure.pi (λ i, μ i))`: when it reads it, it already knows that it appears in a product measure, so it knows that it has to be a measure on the first component of the product, so it knows what `i` has to range over.